goscraper.go

```diff
@@ -449,9 +465,14 @@ func (scraper *Scraper) parseDocument(doc *Document) error {
 		}
 
 		if len(doc.Preview.Title) > 0 && len(doc.Preview.Description) > 0 && ogImage && headPassed {
-			return nil
+			if scraper.Options.MaxTokenDepth == 0 {
+				return nil
+			}
+			if ogType || depth >= scraper.Options.MaxTokenDepth {
+				return nil
+			}
```

Comment:
looking at this code I was thinking that maybe `scrapet.Options.MaxTokenDepth == 0` is redundant, as `depth >= scraper.Options.MaxTokenDepth` will handle it anyway on the first iteration right? So we can have just second `if` 

and another thing which hit me while iterating over this part is that it'll read much better with a little bit different naming, something like `if gotOgType` or `if hasOgType`.

I guess `ogImage` from above could be changed in the same vein to `gotOgImage` or `hasOgImage`?

Reply:
> I guess ogImage from above could be changed in the same vein to gotOgImage or hasOgImage

In origin lib, as you can see - we have `var ogImage bool`. I decide to use same "naming".
btw. Will fix naming in both places. 

> looking at this code I was thinking that maybe `scrapet.Options.MaxTokenDepth == 0` is redundant, as depth >= scraper.Options.MaxTokenDepth will handle it anyway on the first iteration right? 

You are almost right. 
But from **BC** perspective - I'm trying to have same as soon as possible logic that will behave with default value - same as old code. 
The point is that in feature - most probably we will have additional logic/checks after first `if` statement.

Reply:
ohh, that bright future which will come sometime and will force us to change the code! 🙃